app/models/dependency.rb

```diff
@@ -78,12 +78,12 @@ def use_gem_dependency
 
     if gem_dependency.class != Gem::Dependency
       errors.add :rubygem, "Please use Gem::Dependency to specify dependencies."
-      return false
+      throw :abort
     end
 
     if gem_dependency.name.empty?
       errors.add :rubygem, "Blank is not a valid dependency name"
-      return false
+      return :abort
     end
 
     true
```

Comment:
> IMO this change makes it a bit harder to read

@dwradcliffe I agree that change to guard clause was less readable. I had to do it because robocop warned:
>  Style/GuardClause: Use a guard clause instead of wrapping the code inside a conditional expression. (https://github.com/bbatsov/ruby-style-guide#no-nested-conditionals)
    if gem_dependency.name.empty?
    ^^

I have changed it back but it means `true` here is extraneous. Alternatively, we can disable `Style/GuardClause` on this method.